Source/WTF/wtf/cocoa/FileSystemCocoa.mm

```diff
@@ -117,6 +117,14 @@ String extractTemporaryZipArchive(const String& path)
         BOMCopierFree(copier);
     }];
 
+    auto *contentsOfTemporaryDirectory = [NSFileManager.defaultManager contentsOfDirectoryAtPath:temporaryDirectory error:nil];
+    if (contentsOfTemporaryDirectory.count == 1) {
+        auto *subdirectoryPath = [temporaryDirectory stringByAppendingPathComponent:contentsOfTemporaryDirectory.firstObject];
+        BOOL isDirectory;
+        if ([NSFileManager.defaultManager fileExistsAtPath:subdirectoryPath isDirectory:&isDirectory] && isDirectory)
+            temporaryDirectory = subdirectoryPath;
+    }
```

Comment:
You should remove the same code from `WebExtensionCocoa.mm`, as this handles it now.

Reply:
oops, thought i did! must've left that change out. thanks for the catch